api/receiver/receive_results_test.go

```diff
@@ -144,120 +129,203 @@ func TestHandleResultsUpload_extra_params(t *testing.T) {
 	assert.Equal(t, resp.Code, http.StatusOK)
 }
 
-func TestHandleFilePayload(t *testing.T) {
+func TestHandleResultsUpload_single_file(t *testing.T) {
```

Comment:
There's a lot of overlap between this and multiple.
Have a look at `testing.T.Run` for a clean way of collapsing the cases (and we can get rid of  `_single_file` suffixes, yay).